src/subdomains/lightning/services/lightning-transaction.service.ts

```diff
@@ -0,0 +1,278 @@
+    }
+
+    return this.transactionLightningRepo.saveMany(newTransactionEntities);
+  }
+
+  private async getLightningTransactions(
+    action: (startDate: Date, maxTransactions: number, offset: number) => Promise<LndTransactionResponseDto>,
+    startDate: Date,
+    maxTransactions = 1,
+    maxLoop = 100,
+  ): Promise<LndTransactionDto[]> {
+    let offset = 0;
+
+    const allTransactions: LndTransactionDto[] = [];
+
+    for (let i = 0; i < maxLoop; i++) {
+      const transactionsResponse = await action(startDate, maxTransactions, offset);
+      const transactions = transactionsResponse.transactions;
+      allTransactions.push(...transactions);
+
+      if (transactions.length < maxTransactions) {
+        break;
+      }
+
+      offset = transactionsResponse.last_index_offset;
+    }
+
+    if (allTransactions.length === maxTransactions * maxLoop) {
+      throw new Error('possibly further transactions available');
+    }
+
+    return allTransactions;
+  }
+
+  private mapLightningTransactions(
+    type: TransactionLightningType,
+    transactions: LndTransactionDto[],
+    databaseTransactionEntity: TransactionLightningEntity | null,
+  ): TransactionLightningEntity[] {
+    const transactionEntities: TransactionLightningEntity[] = [];
+
+    for (const transaction of transactions) {
+      if (transaction.transaction !== databaseTransactionEntity?.transaction) {
+        const state = <TransactionLightningState>TransactionLightningState[transaction.state];
+        if (!state) throw new NotFoundException(`${type}: Unknown transaction state ${transaction.state}`);
+
+        const paymentRequest = transaction.paymentRequest;
+        const invoiceInfo = paymentRequest ? LightningHelper.getInvoiceInfo(paymentRequest) : undefined;
+
+        const description = transaction.description ?? invoiceInfo?.description;
+        const publicKey = invoiceInfo?.publicKey;
+
+        const transactionEntity = this.transactionLightningRepo.create({
+          type: type,
+          state: state,
+          transaction: transaction.transaction,
+          secret: transaction.secret,
+          publicKey: publicKey,
+          amount: transaction.amount,
+          fee: transaction.fee,
+          creationTimestamp: transaction.creationTimestamp,
+          expiresTimestamp: transaction.expiresTimestamp,
+          confirmedTimestamp: transaction.confirmedTimestamp,
+          description: description,
+          reason: transaction.reason,
+          paymentRequest: paymentRequest,
+        });
+
+        transactionEntities.push(transactionEntity);
+      }
+    }
+
+    return transactionEntities;
+  }
+
+  async syncLightningUserTransactions(): Promise<UserTransactionEntity[]> {
+    const maxId = await this.userTransactionRepository.getMaxId();
+
+    if (!maxId) {
+      return this.saveLightningUserTransactions(null);
+    }
+
+    const databaseUserTransactionEntity = await this.userTransactionRepository.findOneBy({ id: maxId });

```

Comment:
Same here

Reply:
Other implementation, not needed anymore.